### PR TITLE
Add FerryTag 

### PR DESCRIPTION
**Pull request by isabelleh**
This PR adds the Ferry Tag and some of the more common values.

https://taginfo.openstreetmap.org/keys/ferry#overview

### CODE REVIEW

File: src/main/java/org/openstreetmap/atlas/tags/FerryTag.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+import org.openstreetmap.atlas.tags.annotations.Tag;
+import org.openstreetmap.atlas.tags.annotations.TagKey;
+
+/**
+ * OSM Ferry Tag
+ *
+ * @author isabellehillberg
+ */
+@Tag(taginfo = "https://taginfo.openstreetmap.org/keys/ferry#values")
```
Comment (MikeGost) [inline src/main/java/org/openstreetmap/atlas/tags/FerryTag.java]: Nit: Can you link the OSM wiki please:

`osm = "https://wiki.openstreetmap.org/wiki/Key:ferry"`

File: src/main/java/org/openstreetmap/atlas/tags/FerryTag.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    private static final EnumSet<FerryTag> CAV_NAVIGABLE = EnumSet.of(MOTORWAY, TRUNK, PRIMARY,
+            SECONDARY, TERTIARY, UNCLASSIFIED, RESIDENTIAL, SERVICE);
+    private static final EnumSet<FerryTag> PEDESTRIAN_NAVIGABLE = EnumSet.of(FOOTWAY, TRACK,
+            PEDESTRIAN);
+
+    @TagKey
+    public static final String KEY = "ferry";
+
+    public static boolean isCarNavigableHighway(final FerryTag tag)
```
Comment (MikeGost) [inline src/main/java/org/openstreetmap/atlas/tags/FerryTag.java]: Nit: What do you think of `isCarNavigableFerry`?

File: src/main/java/org/openstreetmap/atlas/tags/FerryTag.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    {
+        return CAV_NAVIGABLE.contains(tag);
+    }
+
+    public static boolean isPedestrianNavigableHighway(final FerryTag tag)
```
Comment (MikeGost) [inline src/main/java/org/openstreetmap/atlas/tags/FerryTag.java]: Nit: What do you think of `isPedestrianNavigableFerry`?

File: src/main/java/org/openstreetmap/atlas/tags/FerryTag.java (hunk 1 above)
Comment (isabelleh) [reply src/main/java/org/openstreetmap/atlas/tags/FerryTag.java]: The key is not documented. The route=ferry tag is the official way to tag ferries in OSM, but the ferry tag is used by the community in some cases.

File: src/main/java/org/openstreetmap/atlas/tags/FerryTag.java (hunk 2 above)
Comment (isabelleh) [reply src/main/java/org/openstreetmap/atlas/tags/FerryTag.java]: Yes! copy fail from the highway tag :)

File: src/main/java/org/openstreetmap/atlas/tags/FerryTag.java (hunk 1 above)
Comment (MikeGost) [reply src/main/java/org/openstreetmap/atlas/tags/FerryTag.java]: It is - the wiki above! It's pretty brief but at least gives users a reference point. I'd add it for completeness.

Comment (isabelleh) [reply src/main/java/org/openstreetmap/atlas/tags/FerryTag.java]: Got it!

**Pull request merged**